pygmt/src/grd2xyz.py

```diff
@@ -0,0 +1,106 @@
+"""
+grd2xyz - Convert grid to data table
+"""
+import warnings
+
+import pandas as pd
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    R="region",
+    V="verbose",
+)
+@kwargs_to_strings(R="sequence")
+def grd2xyz(grid, output_type="pandas", outfile=None, **kwargs):
+    r"""
+    Convert grid to data table.
+
+    Read a grid and output xyz-triplets as a :class:`numpy.ndarray`,
+    :class:`pandas.DataFrame`, or ASCII file.
+
+    Full option list at :gmt-docs:`grd2xyz.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a
+        :class:`xarray.DataArray`. This is the only required parameter.
+    output_type : str
+        Determine the format the xyz data will be returned in [Default is
+        ``pandas``]:
+
+            - ``numpy`` - :class:`numpy.ndarray`
+            - ``pandas``- :class:`pandas.DataFrame`
+            - ``file`` - ASCII file (requires ``outfile``)
+    outfile : str
+        The file name for the output ASCII file.
+    {R}
+    {V}
+
+    Returns
+    -------
+    ret : pandas.DataFrame or numpy.ndarray or None
+        Return type depends on ``outfile`` and ``output_type``:
+
+        - None if ``outfile`` is set (output will be stored in file set by
+          ``outfile``)
+        - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile`` is
+          not set (depends on ``output_type`` [Default is
+          :class:`pandas.DataFrame`])
+
+    """
+    if output_type not in ["numpy", "pandas", "file"]:
+        raise GMTInvalidInput(
+            """Must specify format as either numpy, pandas, or file."""
+        )
+    if outfile is not None and output_type != "file":
+        msg = (
+            f"Changing `output_type` of grd2xyz from '{output_type}' to 'file' "
+            "since `outfile` parameter is set. Please use `output_type='file'` "
+            "to silence this warning."
+        )
+        warnings.warn(msg, category=RuntimeWarning, stacklevel=2)
+        output_type = "file"
+    elif output_type == "file" and outfile is None:
+        raise GMTInvalidInput("""Must specify outfile for ASCII output.""")
+    if "o" not in kwargs.keys(): # Only set column names for standard output
+        # Set the default column names for the pandas dataframe header
+        dataframe_header = ["x", "y", "z"]
+        # Set the column names to match an input DataArray as the grid
+        if isinstance(grid, xr.DataArray) and output_type == "pandas":
+            # Reverse the dims because it is rows, columns ordered.
+            dataframe_header = (grid.dims[1],) + (grid.dims[0],) + (grid.name,)
```

Comment:
Does the following code works in the same way?
```suggestion
            dataframe_header = [grid.dims[1], grid.dims[0], grid.name]
```